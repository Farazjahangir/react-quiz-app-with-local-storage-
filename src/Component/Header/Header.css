*{
    margin: 0;
    padding: 0;
}
#header{
    background-color: #34495e;
    padding: 8px;
    display: flex;
    justify-content: space-between;
    width: 100%;
    box-sizing: border-box;
}
.text-white{
    color: white;
}
#header p{
    font-size: 1.4em;
}

@media (max-width : 650px){
    #header p{
        display: none;
    }
    #header h2 , #header h3{
        font-size: 1.3em;
    }
}