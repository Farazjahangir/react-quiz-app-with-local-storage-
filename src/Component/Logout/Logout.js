import React, { Component } from 'react';

class Logout extends Component{
    constructor(props){
        super(props)
    }

    render(){
       const logout = this.props.logoutFunc
        return(
            <div className="text-center my-4">
                <input type="button" value="Logout" className="btn btn-danger" onClick={logout} />
            </div>
        )
    }
}

export default Logout