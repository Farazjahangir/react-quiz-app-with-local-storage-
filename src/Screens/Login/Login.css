form{
    width: 50%;
    margin: 0 auto;
    margin-top: 50px;
}
.text-center{
    text-align: center;
}
.my-1{
    padding-top: 40px;
}