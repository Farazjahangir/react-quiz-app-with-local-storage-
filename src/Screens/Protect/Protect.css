.border{
    border : 1px solid rgba(211, 211, 211, 0.6);
    padding: 45px;
    width: 90%;
    margin: 0 auto;
    background: rgba(211, 211, 211, 0.5)
}