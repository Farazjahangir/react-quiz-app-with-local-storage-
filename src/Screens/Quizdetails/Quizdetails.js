import React, { Component } from 'react';
let quizIndex;
class Quizdetails extends Component{
    constructor(props){
        super(props)
        }

        componentWillMount(){
           quizIndex= localStorage.getItem("selectedQuizIndex")
        }
    render(){
        
        const {quizList , startQuiz} = this.props
        return(
            <div>
                <table className="table">
                    <thead>
                        <tr>
                            <th>Quiz Name</th>
                            <th>Total Questions</th>
                            <th>Total Score</th>
                        </tr>
                    </thead>
                    <tbody>
                        <tr>
                            <td>{quizList[quizIndex].quizName}</td>
                            <td>{quizList[quizIndex].totalQustions }</td>
                            <td>{quizList[quizIndex].totalScore }</td>
                        </tr>
                    </tbody>
                </table>
                <div className="text-center">
                    <input type="button" value="Start Quiz" className="btn btn-success" onClick={startQuiz} />
                </div>
            </div>
        )
    }
}

export default Quizdetails