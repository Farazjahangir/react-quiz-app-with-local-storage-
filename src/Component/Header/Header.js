import React, { Component } from 'react';
import './Header.css';

const Header = ()=>{
    return(
        <div>
            <div id="header">
            <h2 className="text-white">Assignment 5</h2>
            <h3 className="text-white">Quiz App</h3>
            <p  className="text-white">Faraz Jahangir</p>
            </div>
        </div>
    )
}
export default Header;
