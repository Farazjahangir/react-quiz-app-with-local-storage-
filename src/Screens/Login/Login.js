import React, { Component } from 'react';
import './Login.css';

class Login extends Component{
    constructor(props){
        super(props)
        console.log("Signin " ,props);
        
        this.state = {
            email : '',
            password : ''
        }
    }

    render() {
       const {showSignupForm , signInFunc} = this.props
       const {email , password} = this.state
        return( 
        <div>
            <h2 className="text-center my-1">Sign In To Your Account</h2>
            <form>
                <div className="form-group">
                    <label htmlFor="email">Email address:</label>
                    <input type="email" value ={email} className="form-control" id="email" onChange={(e)=>{this.setState({email : e.target.value})}} />
                </div>
                <div className="form-group">
                    <label htmlFor="pwd">Password:</label>
                    <input type="password" value={password} className="form-control" id="pwd" onChange={(e)=>{this.setState({password : e.target.value})}} />
                </div>   
                <div className="form-group">
                <input type="Button" value="SignIn" className="btn btn-danger mx-1" onClick={()=>{signInFunc(email , password)}} />    
                <input type="button" value="Signup" className="btn btn-success" onClick={showSignupForm} />    
                </div>
            </form>
        </div>
    )
    }
}
export default Login;