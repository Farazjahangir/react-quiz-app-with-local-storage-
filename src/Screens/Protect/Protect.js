import React, { Component } from 'react';
import "./Protect.css"
class Protect extends Component{
    constructor(props){
        super(props)
            this.state = {
                key : ''
            }
        }

    render(){
        const {submitKeyFunc} = this.props
        const {key} = this.state
        return(
            <div className="input-group my-5 border">
                <input type="password" value={key} placeholder="Enter Proctoring Key" className="form-control" onChange={(e)=>{this.setState({key : e.target.value})}} />
                <input type="button" value="Submit" className="btn btn-danger" onClick={()=>{submitKeyFunc(key)}} />
            </div>
        )
    }
    }

export default Protect