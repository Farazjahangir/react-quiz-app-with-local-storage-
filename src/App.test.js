import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Header from './Component/Header/Header';
import Logout from './Component/Logout/Logout';
import Login from './Screens/Login/Login';
import Protect from './Screens/Protect/Protect';
import Quizdetails from './Screens/Quizdetails/Quizdetails';
import Quizpage from './Screens/Quizpage/Quizpage';
import Quizzes from './Screens/Quizzes/Quizzes';
import Signup from './Screens/Signup/Signup';

Enzyme.configure({ adapter: new Adapter() });


describe('Header', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Header />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    test('has a valid snapshot', () => {
        const component = renderer.create(
            <Header />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});



describe('Logout', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Logout />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    test('has a valid snapshot', () => {
        const component = renderer.create(
            <Logout />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('Login', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Login />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    test('has a valid snapshot', () => {
        const component = renderer.create(
            <Login />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});




describe('Protect', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Protect />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    test('has a valid snapshot', () => {
        const component = renderer.create(
            <Protect />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});




describe('Quizdetails', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Quizdetails />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    test('has a valid snapshot', () => {
        const component = renderer.create(
            <Quizdetails />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});




describe('Quizpage', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Quizpage />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    test('has a valid snapshot', () => {
        const component = renderer.create(
            <Quizpage />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});



describe('Quizzes', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Quizzes />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    test('has a valid snapshot', () => {
        const component = renderer.create(
            <Quizzes />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});




describe('Signup', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Signup />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    test('has a valid snapshot', () => {
        const component = renderer.create(
            <Signup />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});