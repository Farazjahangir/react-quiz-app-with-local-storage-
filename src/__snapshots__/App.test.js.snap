// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header has a valid snapshot 1`] = `
<div>
  <div
    id="header"
  >
    <h2
      className="text-white"
    >
      Assignment 5
    </h2>
    <h3
      className="text-white"
    >
      Quiz App
    </h3>
    <p
      className="text-white"
    >
      Faraz Jahangir
    </p>
  </div>
</div>
`;

exports[`Login has a valid snapshot 1`] = `
<div>
  <h2
    className="text-center my-1"
  >
    Sign In To Your Account
  </h2>
  <form>
    <div
      className="form-group"
    >
      <label
        htmlFor="email"
      >
        Email address:
      </label>
      <input
        className="form-control"
        id="email"
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="pwd"
      >
        Password:
      </label>
      <input
        className="form-control"
        id="pwd"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="btn btn-danger mx-1"
        onClick={[Function]}
        type="Button"
        value="SignIn"
      />
      <input
        className="btn btn-success"
        onClick={undefined}
        type="button"
        value="Signup"
      />
    </div>
  </form>
</div>
`;

exports[`Login has a valid snapshot 2`] = `
<div>
  <h2
    className="text-center my-1"
  >
    Sign In To Your Account
  </h2>
  <form>
    <div
      className="form-group"
    >
      <label
        htmlFor="email"
      >
        Email address:
      </label>
      <input
        className="form-control"
        id="email"
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="pwd"
      >
        Password:
      </label>
      <input
        className="form-control"
        id="pwd"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="btn btn-danger mx-1"
        onClick={[Function]}
        type="Button"
        value="SignIn"
      />
      <input
        className="btn btn-success"
        onClick={undefined}
        type="button"
        value="Signup"
      />
    </div>
  </form>
</div>
`;

exports[`Login has a valid snapshot 3`] = `
<div>
  <h2
    className="text-center my-1"
  >
    Sign In To Your Account
  </h2>
  <form>
    <div
      className="form-group"
    >
      <label
        htmlFor="email"
      >
        Email address:
      </label>
      <input
        className="form-control"
        id="email"
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="pwd"
      >
        Password:
      </label>
      <input
        className="form-control"
        id="pwd"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="btn btn-danger mx-1"
        onClick={[Function]}
        type="Button"
        value="SignIn"
      />
      <input
        className="btn btn-success"
        onClick={undefined}
        type="button"
        value="Signup"
      />
    </div>
  </form>
</div>
`;

exports[`Login has a valid snapshot 4`] = `
<div>
  <h2
    className="text-center my-1"
  >
    Sign In To Your Account
  </h2>
  <form>
    <div
      className="form-group"
    >
      <label
        htmlFor="email"
      >
        Email address:
      </label>
      <input
        className="form-control"
        id="email"
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="pwd"
      >
        Password:
      </label>
      <input
        className="form-control"
        id="pwd"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="btn btn-danger mx-1"
        onClick={[Function]}
        type="Button"
        value="SignIn"
      />
      <input
        className="btn btn-success"
        onClick={undefined}
        type="button"
        value="Signup"
      />
    </div>
  </form>
</div>
`;

exports[`Login has a valid snapshot 5`] = `
<div>
  <h2
    className="text-center my-1"
  >
    Sign In To Your Account
  </h2>
  <form>
    <div
      className="form-group"
    >
      <label
        htmlFor="email"
      >
        Email address:
      </label>
      <input
        className="form-control"
        id="email"
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="pwd"
      >
        Password:
      </label>
      <input
        className="form-control"
        id="pwd"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="btn btn-danger mx-1"
        onClick={[Function]}
        type="Button"
        value="SignIn"
      />
      <input
        className="btn btn-success"
        onClick={undefined}
        type="button"
        value="Signup"
      />
    </div>
  </form>
</div>
`;

exports[`Login has a valid snapshot 6`] = `
<div>
  <h2
    className="text-center my-1"
  >
    Sign In To Your Account
  </h2>
  <form>
    <div
      className="form-group"
    >
      <label
        htmlFor="email"
      >
        Email address:
      </label>
      <input
        className="form-control"
        id="email"
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="pwd"
      >
        Password:
      </label>
      <input
        className="form-control"
        id="pwd"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="btn btn-danger mx-1"
        onClick={[Function]}
        type="Button"
        value="SignIn"
      />
      <input
        className="btn btn-success"
        onClick={undefined}
        type="button"
        value="Signup"
      />
    </div>
  </form>
</div>
`;

exports[`Login has a valid snapshot 7`] = `
<div>
  <h2
    className="text-center my-1"
  >
    Sign In To Your Account
  </h2>
  <form>
    <div
      className="form-group"
    >
      <label
        htmlFor="email"
      >
        Email address:
      </label>
      <input
        className="form-control"
        id="email"
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="pwd"
      >
        Password:
      </label>
      <input
        className="form-control"
        id="pwd"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="btn btn-danger mx-1"
        onClick={[Function]}
        type="Button"
        value="SignIn"
      />
      <input
        className="btn btn-success"
        onClick={undefined}
        type="button"
        value="Signup"
      />
    </div>
  </form>
</div>
`;

exports[`Logout has a valid snapshot 1`] = `
<div
  className="text-center my-4"
>
  <input
    className="btn btn-danger"
    onClick={undefined}
    type="button"
    value="Logout"
  />
</div>
`;

exports[`Protect has a valid snapshot 1`] = `
<div
  className="input-group my-5 border"
>
  <input
    className="form-control"
    onChange={[Function]}
    placeholder="Enter Proctoring Key"
    type="password"
    value=""
  />
  <input
    className="btn btn-danger"
    onClick={[Function]}
    type="button"
    value="Submit"
  />
</div>
`;

exports[`Signup has a valid snapshot 1`] = `
<div>
  <h2
    className="text-center my-1"
  >
    Sign up
  </h2>
  <form>
    <div
      className="form-group"
    >
      <label
        htmlFor="username"
      >
        Username:
      </label>
      <input
        className="form-control"
        id="username"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="email"
      >
        Email address:
      </label>
      <input
        className="form-control"
        id="email"
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="pwd"
      >
        Password:
      </label>
      <input
        className="form-control"
        id="pwd"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <input
      className="btn btn-danger mx-1"
      onClick={[Function]}
      type="button"
      value="Submit"
    />
    <input
      className="btn btn-primary"
      onClick={undefined}
      type="button"
      value="SignIn"
    />
  </form>
</div>
`;
